packages/clerk-js/src/ui/signUp/signUpFormHelpers.ts

```diff
@@ -124,47 +146,73 @@ function getField(fieldKey: FieldKey, fieldProps: FieldDeterminationProps): Fiel
   }
 }
 
-// TODO revisit when attributes.email_address.required becomes effective
 function getEmailAddressField({
   attributes,
   hasTicket,
   hasEmail,
   activeCommIdentifierType,
+  isProgressiveSignUp,
 }: FieldDeterminationProps): Field | undefined {
+  if (isProgressiveSignUp) {
+    // If there is no ticket, or there is a ticket along with an email, and email address is enabled,
+    // we have to show it in the SignUp form
+    let show = (!hasTicket || (hasTicket && hasEmail)) && attributes.email_address.enabled;
+    // If we are in the case of Email OR Phone, determine if the initial input has to be the email address
+    // based on the active identifier type.
+    if (emailOrPhone(attributes, isProgressiveSignUp)) {
+      show = show && activeCommIdentifierType === 'emailAddress';
+    }
+
+    return {
+      required: attributes.email_address.required,
+      disabled: !!hasTicket && !!hasEmail,
+      show,
+    };
+  }
+
   const show =
     (!hasTicket || (hasTicket && hasEmail)) &&
     attributes.email_address.enabled &&
     attributes.email_address.used_for_first_factor &&
     activeCommIdentifierType == 'emailAddress';
 
-  if (!show) {
-    return;
-  }
-
   return {
     required: true, // as far as the FE is concerned the email address is required, if shown
     disabled: !!hasTicket && !!hasEmail,
+    show,
   };
 }
 
-// TODO revisit when attributes.phone_number.required becomes effective
 function getPhoneNumberField({
   attributes,
   hasTicket,
   activeCommIdentifierType,
+  isProgressiveSignUp,
 }: FieldDeterminationProps): Field | undefined {
```

Comment:
❓ 
Can this in any way return undefined ?

Reply:
Nice catch, fixed